### PR TITLE
RD-1514 - feat: wrap only once

### PR DESCRIPTION
**Pull request by saartochner**

### CODE REVIEW

File: src/test/unit/sync_http/test_sync_hook.py [hunk 1]
```diff
@@ -364,8 +364,8 @@ def lambda_test_function(event, context):
 
     assert lambda_test_function({}, SimpleNamespace(aws_request_id="1234")) == 1
     assert Configuration.enhanced_print is True
-    assert "RequestId: 1234 test_sync_hook.py" in caplog.text
```
Comment (saartochner) [inline src/test/unit/sync_http/test_sync_hook.py]: these changes are because of the pytest upgrade....

File: src/lumigo_tracer/__init__.py [hunk 2]
```diff
@@ -1 +1,2 @@
 from .sync_http.sync_hook import lumigo_tracer, LumigoChalice  # noqa
+from .sync_http.handler import handler  # noqa
```
Comment (nirLumigo) [inline src/lumigo_tracer/__init__.py]: Maybe it's better to export it as `_handler`?
The user won't call it and it's only for internal usage.

File: src/lumigo_tracer/sync_http/handler.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        )
+    except KeyError:
+        raise ValueError(
+            "Could not find the original handler. Please follow lumigo's docs: https://docs.lumigo.io/"
+        )
+    except Exception:
+        raise Exception(
+            "Problem occurred in lumigo's wrapper. Make sure that you followed the docs or contact us."
+        )
+    original_handler(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (saartochner) [reply src/lumigo_tracer/sync_http/handler.py]: 😟

File: src/lumigo_tracer/sync_http/sync_hook.py [hunk 4]
```diff
@@ -127,12 +128,33 @@ def _putheader_wrapper(func, instance, args, kwargs):
     return ret_val
 
 
+def _context_already_wrapped(*args):
+    """
+    This function is here in order to validate that we didn't already wrap this invocation
+        (using the sls plugin / auto instrumentation / etc.)
+    """
+    return len(args) >= 2 and hasattr(args[1], CONTEXT_WRAPPED_BY_LUMIGO_KEY)
```
[the remark this answers is not in the document]
Comment (saartochner) [reply src/lumigo_tracer/sync_http/sync_hook.py]: yes,
but I want to be robust for cases that the user use it in weird ways and passes fewer/more arguments

File: src/test/unit/sync_http/test_sync_hook.py [hunk 5]
```diff
@@ -477,3 +484,14 @@ def lambda_test_function(event, context):
...
+
+def test_can_not_wrap_twice(reporter_mock):
+    @lumigo_tracer()
+    @lumigo_tracer()
+    def lambda_test_function(event, context):
+        return "ret_value"
+
+    result = lambda_test_function({}, SimpleNamespace(aws_request_id="1234"))
+    assert result == "ret_value"
+    assert reporter_mock.call_count == 2
```
[the remark this answers is not in the document]
Comment (saartochner) [reply src/test/unit/sync_http/test_sync_hook.py]: start and end.. :)

File: src/lumigo_tracer/sync_http/handler.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import os
+from importlib import import_module
+
+from lumigo_tracer import lumigo_tracer
+
+ORIGINAL_HANDLER_KEY = "LUMIGO_ORIGINAL_HANDLER"
+
+
+@lumigo_tracer()
+def handler(*args, **kwargs):
```
[the remark this answers is not in the document]
Comment (saartochner) [reply src/lumigo_tracer/sync_http/handler.py]: only from the auto-instrument,
where we change the handler of the lambda into this handler :)

**Pull request merged**